include/pqxx/internal/header-pre.hxx

```diff
@@ -58,6 +58,18 @@
 // Workarounds & definitions that need to be included even in library's headers
 #include "pqxx/config-public-compiler.h"
 
+#if defined(_MSC_VER)
+#  define PQXX_CPLUSPLUS _MSVC_LANG
```

Comment:
Why this?  Surely MSVC defines `__cplusplus` like any other compiler?

Reply:
In the MSVC compiler, this is because the `__cplusplus` macro only returns `199711` (unaffected by the C++ standard).
Use `_MSVC_LANG` instead.
By specifying the compiler option `/Zc:__cplusplus`, `__cplusplus` will return the C++ standard version just like other compilers, but there is no guarantee that it is always specified.

See also:
  - https://stackoverflow.com/questions/37503029/cplusplus-is-equal-to-199711-in-msvc-does-it-support-c11
  - https://stackoverflow.com/questions/43639322/is-it-safe-to-use-msvc-lang-instead-of-cplusplus
  - https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170

Reply:
> In the MSVC compiler, this is because the `__cplusplus` macro only returns `199711` (unaffected by the C++ standard). Use `_MSVC_LANG` instead. By specifying the compiler option `/Zc:__cplusplus`, `__cplusplus` will return the C++ standard version just like other compilers, but there is no guarantee that it is always specified.

Thanks for explaining.  I'm feeling a little exasperated with this compiler today...